xbmc/platform/freebsd/CPUInfoFreebsd.cpp

```diff
@@ -135,3 +239,20 @@ float CCPUInfoFreebsd::GetCPUFrequency()
 
   return static_cast<float>(hz);
 }
+
+
+bool CCPUInfoFreebsd::GetTemperature(CTemperature& temperature)
+{
+  int value;
+  size_t len;
+
+  len = sizeof(value);
+  if (sysctlbyname("dev.cpu.0.temperature", &value, &len, NULL, 0) != 0)
+    return false;
+  /* temperature is in kelvin * 10 */
+  value = ((value - 2732) / 10);
+  temperature = CTemperature::CreateFromCelsius(value);
```

Comment:
if the temp is returned in kelvin you can just create it from kelvin instead of converting yourself. https://github.com/xbmc/xbmc/blob/master/xbmc/utils/Temperature.h#L34

Reply:
Sounds good, but an implementation for CreateFromKelvin was missing. I had to add one first.